### PR TITLE
indexes to reducers

### PR DESCRIPTION
**Pull request by rfornea**
Pulls those indexes into reducers.  You're not supposed to put Math.Random() in a reducer apparently, so I do that calculation outside of the reducer and just send to the reducer for the actual setting.

### CODE REVIEW

File: src/services/iota.js [hunk 1]
```diff
@@ -72,69 +65,67 @@ const skinnyQueryTransactions = (iotaProvider, addresses) =>
...
   });
 
   let frontOfFile = new Promise((resolve, reject) => {
-    skinnyQueryTransactions(IotaA, [addresses[indexes.frontIdx]]).then(
+    skinnyQueryTransactions(IotaA, [addresses[idx.frontIdx]]).then(
       transactions => {
         if (transactions.length > 0) {
-          indexes.latestFoundFrontIdx = indexes.frontIdx;
-          updateFrontIndex(indexes.frontIdx);
+          store.dispatch(
```
Comment (AaronVasquez) [inline src/services/iota.js]: Dispatch this from an epic after the promise resolves. Files in `/services` shouldn't have any dependencies from our business logic code.

File: src/services/iota.js [hunk 2]
```diff
@@ -72,69 +65,67 @@ const skinnyQueryTransactions = (iotaProvider, addresses) =>
...
 
   return Promise.all([frontOfFile, backOfFile]).then(() => {
-    return recalculatePercentage(indexes);
+    return recalculatePercentage(store.getState().upload);
   });
 };
 
-const recalculatePercentage = indexes => {
-  if (indexes.latestFoundFrontIdx >= indexes.latestFoundBackIdx - 1) {
+const recalculatePercentage = uploadState => {
```
Comment (EdmundMai) [inline src/services/iota.js]: pass as separate args instead of depending on state

**Comment by rfornea** [comment]
Don't look at this again yet.  Still have to make a few more changes.

File: src/services/iota.js [hunk 3]
```diff
@@ -72,69 +59,40 @@ const skinnyQueryTransactions = (iotaProvider, addresses) =>
...
   });
 
-  return Promise.all([frontOfFile, backOfFile]).then(() => {
-    return recalculatePercentage(indexes);
+  return Promise.all([frontOfFile, backOfFile]).then(indexResults => {
+    return {
+      frontIndex: indexResults[0].frontIndex,
+      updateFrontIndex: indexResults[0].updateIndex,
+      backIndex: indexResults[1].backIndex,
+      updateBackIndex: indexResults[1].updateIndex
```
Comment (EdmundMai) [inline src/services/iota.js]: you could do `const [a, b] = indexResults;` (name it something other than a and b of course) and then just do `a. frontIndex` etc. no biggie though just letting you know

File: src/redux/reducers/upload-reducer.js [hunk 4]
```diff
@@ -77,6 +95,20 @@ const uploadReducer = (state = initState, action) => {
...
+      const { frontIdx, backIdx, dataMapLength } = action.payload;
+      const indexes = {
+        ...state.indexes,
+        endingIdx: dataMapLength - 1,
+        frontIdx,
+        backIdx
+      };
+      return {
+        ...state,
+        indexes: indexes,
```
Comment (EdmundMai) [inline src/redux/reducers/upload-reducer.js]: just `indexes` works too, it's shorthand for `indexes: indexes`

Comment (EdmundMai) [reply src/redux/reducers/upload-reducer.js]: or if you want to shrink it even more, just take out the variable assignment altogether

```
+      indexes: {
+        ...state.indexes,
+        endingIdx: dataMapLength - 1,
+        frontIdx,
+        backIdx
+      };
```

File: src/redux/reducers/upload-reducer.js [hunk 5]
```diff
@@ -34,13 +41,24 @@ const uploadReducer = (state = initState, action) => {
...
+        backIndex
+      } = action.payload;
       const updatedHistory = state.history.map(f => {
         return f.handle === fileHandle ? { ...f, uploadProgress } : f;
       });
+      const newIndexes = {
+        ...state.indexes,
+        frontIdx: frontIndex,
+        backIdx: backIndex
+      };
```
Comment (EdmundMai) [inline src/redux/reducers/upload-reducer.js]: same for this

**Pull request merged**